lib/spack/spack/cmd/create.py

```diff
@@ -120,39 +138,78 @@ def setup_parser(subparser):
 
 class ConfigureGuesser(object):
     def __call__(self, stage):
-        """Try to guess the type of build system used by the project, and return
-           an appropriate configure line.
-        """
-        autotools = "configure('--prefix=%s' % prefix)"
-        cmake     = "cmake('.', *std_cmake_args)"
-        python    = "python('setup.py', 'install', '--prefix=%s' % prefix)"
-        r         = "R('CMD', 'INSTALL', '--library=%s' % self.module.r_lib_dir, '%s' % self.stage.archive_file)"
-
-        config_lines = ((r'/configure$',      'autotools', autotools),
-                        (r'/CMakeLists.txt$', 'cmake',     cmake),
-                        (r'/setup.py$',       'python',    python),
-                        (r'/NAMESPACE$',      'r',         r))
-
-        # Peek inside the tarball.
-        tar = which('tar')
-        output = tar(
-            "--exclude=*/*/*", "-tf", stage.archive_file, output=str)
-        lines = output.split("\n")
-
-        # Set the configure line to the one that matched.
-        for pattern, bs, cl in config_lines:
+        """Try to guess the type of build system used by the project. Set the
+        appropriate default configure, build, and install instructions."""
+
+        # Default configure instructions
+        configureDict = {
+            'autotools': "configure('--prefix={0}'.format(prefix))",
+            'cmake':     "cmake('.', *std_cmake_args)",
+            'scons':     "",
+            'python':    "",
+            'r':         "",
+            'unknown':   "# FIXME: Unknown build system"
+        }
+
+        # Default build instructions
+        buildDict = {
+            'autotools': "make()",
+            'cmake':     "make()",
+            'scons':     "scons('prefix={0}'.format(prefix))",
+            'python':    "",
+            'r':         "",
+            'unknown':   "make()",
+        }
+
+        # Default install instructions
+        installDict = {
+            'autotools': "make('install')",
+            'cmake':     "make('install')",
+            'scons':     "scons('install')",
+            'python':    "python('setup.py', 'install', " +
+                         "'--prefix={0}'.format(prefix))",
+            'r':         "R('CMD', 'INSTALL', '--library={0}'.format(" +
+                         "self.module.r_lib_dir), self.stage.archive_file)",
+            'unknown':   "make('install')",
+        }
```

Comment:
Honestly might just be better to combine all of these into a general "install instructions" dictionary, where each entry contains all the lines for configure, build, and install. That way, we can personalize the comments, and there won't be empty lines everywhere. Let me know if anyone thinks this is a good idea.